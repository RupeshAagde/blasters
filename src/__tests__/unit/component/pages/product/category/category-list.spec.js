import {mount} from "@vue/test-utils";
import * as sinon from "sinon";
import { console, Promise } from "window-or-global";
import EditCategory from "../../../../../../pages/product/category/list.vue";
import URLS from  "../../../../../../services/company-admin.service"


describe('EditCategory', () => {
it('has data', () => {
	expect(typeof EditCategory.data).toBe('function');
})
})

const stubMethod = sinon.stub(URLS,"fetchCategory_v2");

describe('Mounted Category List', () => {
    let wrapper;
    beforeAll(() => {
        console.log("**********************",stubMethod)
        stubMethod.callsFake(() => Promise.resolve([{}]))
        wrapper = mount(EditCategory,{
            propsData: {
            },
			mocks:{
                $route: {
                    params: {}
				}
			}
		}
        );
    })
    it('has data', () => {
        expect(typeof EditCategory.data).toBe('function');
    })
    
	test('is a Vue instance', () => {
	  expect(wrapper.vm).toBeTruthy()
	})
	// it('has a button', () => {
	// 	expect(wrapper.find('image-uploader-tile')).toBe(true)
	// })
	it('should render to a snapshot', () => {
		expect(wrapper.element).toMatchSnapshot()
	  })
  }
)
