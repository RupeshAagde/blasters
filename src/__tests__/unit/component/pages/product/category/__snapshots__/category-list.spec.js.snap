// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mounted Category List should render to a snapshot 1`] = `
<div
  class="main-container"
>
  <div>
    <div
      class="jumbotron-container"
    >
      <div
        class="txt-box"
      >
        <div
          class="jumbotron-title"
        >
          Category
        </div>
         
        <div
          class="nitrozen-tooltip cst-space-left"
        >
          <span
            class="nitrozen-inline-svg"
          >
            <svg
              height="12px"
              version="1.1"
              viewBox="0 0 12 12"
              width="12px"
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            >
              
                    
              <desc>
                Created with sketchtool.
              </desc>
              
                    
              <g
                fill="none"
                fill-rule="evenodd"
                id="Components"
                stroke="none"
                stroke-width="1"
              >
                
                        
                <g
                  fill="#41434C"
                  fill-rule="nonzero"
                  id="Text-fields"
                  transform="translate(-312.000000, -1846.000000)"
                >
                  
                            
                  <g
                    id="Group-9"
                    transform="translate(306.000000, 1840.000000)"
                  >
                    
                                
                    <g
                      id="noun_Information_55404"
                      transform="translate(6.000000, 6.000000)"
                    >
                      
                                    
                      <g
                        id="Group"
                        transform="translate(5.000000, 3.000000)"
                      >
                        
                                        
                        <path
                          d="M1.31973333,1.37536 C1.55773333,1.37536 1.75053333,1.31662222 1.89266667,1.20088889 C2.04053333,1.08067556 2.1156,0.918026667 2.1156,0.717297778 C2.1156,0.516817778 2.04066667,0.354044444 1.8928,0.233831111 C1.7504,0.117848889 1.55746667,0.0592355556 1.31973333,0.0592355556 C1.09506667,0.0592355556 0.906133333,0.117351111 0.758133333,0.231964444 C0.602666667,0.352302222 0.524,0.515448889 0.524,0.717297778 C0.524,0.919271111 0.6028,1.08266667 0.758133333,1.20263111 C0.905866667,1.31724444 1.09493333,1.37536 1.31973333,1.37536 Z"
                          id="Path"
                        />
                        
                                        
                        <polygon
                          id="Path"
                          points="1.94306667 5.98839111 1.94306667 1.96497778 0.052 1.96497778 0.052 2.63735111 0.723466667 2.63735111 0.723466667 5.98839111 0.052 5.98839111 0.052 6.66076444 2.61466667 6.66076444 2.61466667 5.98839111"
                        />
                        
                                    
                      </g>
                      
                                    
                      <path
                        d="M6,0.666666667 C8.9408,0.666666667 11.3333333,3.0592 11.3333333,6 C11.3333333,8.9408 8.9408,11.3333333 6,11.3333333 C3.0592,11.3333333 0.666666667,8.9408 0.666666667,6 C0.666666667,3.0592 3.0592,0.666666667 6,0.666666667 M6,0 C2.68626667,0 0,2.68626667 0,6 C0,9.31373333 2.68626667,12 6,12 C9.31373333,12 12,9.31373333 12,6 C12,2.68626667 9.31373333,0 6,0 L6,0 Z"
                        id="Shape"
                      />
                      
                                
                    </g>
                    
                            
                  </g>
                  
                        
                </g>
                
                    
              </g>
              
                    
            </svg>
          </span>
          <span
            class="nitrozen-tooltiptext nitrozen-tooltip-bottom"
          >
            <div
              class="custom-tooltip-txt"
            >
              Manage or create categories
            </div>
          </span>
        </div>
      </div>
       
      <div>
        <button
          class="n-button ripple n-button-secondary n-flat-button n-flat-button-secondary"
          href=""
          type="button"
        >
          <div
            class="n-button-content"
          >
            <!---->
            Create Category
            <!---->
          </div>
        </button>
         
      </div>
    </div>
  </div>
   
  <div
    class="search-filter"
  >
    <div
      class="nitrozen-form-input search"
    >
      <div
        class="n-input-label-container"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="nitrozen-input-grp"
      >
        <span
          class="nitrozen-search-icon"
        >
          <span
            class="nitrozen-inline-svg"
          >
            <svg
              height="14px"
              version="1.1"
              viewBox="0 0 14 14"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            >
              
                
              <desc>
                Created with sketchtool.
              </desc>
              
                
              <g
                fill="none"
                fill-rule="evenodd"
                id="Components"
                stroke="none"
                stroke-width="1"
              >
                
                    
                <g
                  fill="#41434C"
                  fill-rule="nonzero"
                  id="Text-fields"
                  transform="translate(-167.000000, -1346.000000)"
                >
                  
                        
                  <g
                    id="Group-2"
                    transform="translate(149.000000, 1333.000000)"
                  >
                    
                            
                    <g
                      id="search-black"
                      transform="translate(18.000000, 13.000000)"
                    >
                      
                                
                      <path
                        d="M9.7848573,8.70634175 C9.8719311,8.74598369 9.953344,8.80145225 10.0245169,8.87262516 L13.7577948,12.605903 C14.0785101,12.9266183 14.082497,13.4426136 13.7617395,13.763371 C13.4432029,14.0819077 12.921818,14.0769729 12.6042714,13.7594263 L8.87099363,10.0261485 C8.79956965,9.9547245 8.7438542,9.8736154 8.70393584,9.7871344 C7.79450938,10.4714775 6.66349636,10.8770903 5.43776104,10.8770903 C2.43456855,10.8770903 0,8.44217072 0,5.43854516 C0,2.43491961 2.43456855,0 5.43776104,0 C8.44095354,0 10.8755221,2.43491961 10.8755221,5.43854516 C10.8755221,6.66497431 10.4696255,7.79658757 9.7848573,8.70634175 Z M5.43776104,9.7893813 C7.84031504,9.7893813 9.7879699,7.84144561 9.7879699,5.43854516 C9.7879699,3.03564472 7.84031504,1.08770903 5.43776104,1.08770903 C3.03520705,1.08770903 1.08755221,3.03564472 1.08755221,5.43854516 C1.08755221,7.84144561 3.03520705,9.7893813 5.43776104,9.7893813 Z"
                        id="Search-Icon"
                      />
                      
                            
                    </g>
                    
                        
                  </g>
                  
                    
                </g>
                
                
              </g>
              
                
            </svg>
          </span>
        </span>
        <!---->
        <input
          class="n-input input-text nitrozen-search-input-padding"
          id="nitrozen-inputk3r6lzd0t"
          max="0"
          min="0"
          placeholder="Search by name..."
          type="search"
        />
        <!---->
        <!---->
      </div>
    </div>
     
    <div
      class="filter"
    >
      <div
        class="filter-item"
      >
        <label
          class="label"
        >
          Level
        </label>
         
        <div
          class="nitrozen-dropdown-container filter-dropdown"
        >
          <!---->
          <div
            class="nitrozen-select-wrapper"
          >
            <div
              class="nitrozen-select"
            >
              <div
                class="nitrozen-select__trigger"
              >
                <span>
                  All
                </span>
                <div
                  class="nitrozen-dropdown-arrow"
                >
                  <span
                    class="nitrozen-inline-svg"
                  >
                    <svg
                      height="24px"
                      version="1.1"
                      viewBox="0 0 24 24"
                      width="24px"
                      xmlns="http://www.w3.org/2000/svg"
                      xmlns:xlink="http://www.w3.org/1999/xlink"
                    >
                      
                                
                      <!-- Generator: Sketch 63.1 (92452) - https://sketch.com -->
                      
                                
                      <title>
                        DropDown Arrow Down
                      </title>
                      
                                
                      <desc>
                        Created with Sketch.
                      </desc>
                      
                                
                      <g
                        fill="none"
                        fill-rule="evenodd"
                        id="Components"
                        stroke="none"
                        stroke-width="1"
                      >
                        
                                    
                        <g
                          id="Group-8"
                        >
                          
                                        
                          <rect
                            fill="#FFFFFF"
                            fill-opacity="0.01"
                            height="24"
                            id="Rectangle"
                            width="24"
                            x="0"
                            y="0"
                          />
                          
                                        
                          <polygon
                            fill="#5C6BDD"
                            id="Shape"
                            points="8 10 12 14 16 10"
                          />
                          
                                    
                        </g>
                        
                                
                      </g>
                      
                            
                    </svg>
                  </span>
                </div>
              </div>
              <div
                class="nitrozen-options"
              >
                <span
                  class="nitrozen-option ripple selected"
                  data-value="all"
                >
                  <div
                    class="nitrozen-option-container"
                  >
                    <span
                      class="nitrozen-option-image"
                    >
                      <!---->
                       All 
                    </span>
                  </div>
                </span>
                <span
                  class="nitrozen-option ripple"
                  data-value="1"
                >
                  <div
                    class="nitrozen-option-container"
                  >
                    <span
                      class="nitrozen-option-image"
                    >
                      <!---->
                       Level 1 
                    </span>
                  </div>
                </span>
                <span
                  class="nitrozen-option ripple"
                  data-value="2"
                >
                  <div
                    class="nitrozen-option-container"
                  >
                    <span
                      class="nitrozen-option-image"
                    >
                      <!---->
                       Level 2 
                    </span>
                  </div>
                </span>
                <span
                  class="nitrozen-option ripple"
                  data-value="3"
                >
                  <div
                    class="nitrozen-option-container"
                  >
                    <span
                      class="nitrozen-option-image"
                    >
                      <!---->
                       Level 3 
                    </span>
                  </div>
                </span>
                <!---->
              </div>
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="filter-item"
      >
        <label
          class="label"
        >
          Status
        </label>
         
        <div
          class="nitrozen-dropdown-container filter-dropdown"
        >
          <!---->
          <div
            class="nitrozen-select-wrapper"
          >
            <div
              class="nitrozen-select"
            >
              <div
                class="nitrozen-select__trigger"
              >
                <span>
                  All
                </span>
                <div
                  class="nitrozen-dropdown-arrow"
                >
                  <span
                    class="nitrozen-inline-svg"
                  >
                    <svg
                      height="24px"
                      version="1.1"
                      viewBox="0 0 24 24"
                      width="24px"
                      xmlns="http://www.w3.org/2000/svg"
                      xmlns:xlink="http://www.w3.org/1999/xlink"
                    >
                      
                                
                      <!-- Generator: Sketch 63.1 (92452) - https://sketch.com -->
                      
                                
                      <title>
                        DropDown Arrow Down
                      </title>
                      
                                
                      <desc>
                        Created with Sketch.
                      </desc>
                      
                                
                      <g
                        fill="none"
                        fill-rule="evenodd"
                        id="Components"
                        stroke="none"
                        stroke-width="1"
                      >
                        
                                    
                        <g
                          id="Group-8"
                        >
                          
                                        
                          <rect
                            fill="#FFFFFF"
                            fill-opacity="0.01"
                            height="24"
                            id="Rectangle"
                            width="24"
                            x="0"
                            y="0"
                          />
                          
                                        
                          <polygon
                            fill="#5C6BDD"
                            id="Shape"
                            points="8 10 12 14 16 10"
                          />
                          
                                    
                        </g>
                        
                                
                      </g>
                      
                            
                    </svg>
                  </span>
                </div>
              </div>
              <div
                class="nitrozen-options"
              >
                <span
                  class="nitrozen-option ripple selected"
                  data-value="all"
                >
                  <div
                    class="nitrozen-option-container"
                  >
                    <span
                      class="nitrozen-option-image"
                    >
                      <!---->
                       All 
                    </span>
                  </div>
                </span>
                <span
                  class="nitrozen-option ripple"
                  data-value="true"
                >
                  <div
                    class="nitrozen-option-container"
                  >
                    <span
                      class="nitrozen-option-image"
                    >
                      <!---->
                       Active 
                    </span>
                  </div>
                </span>
                <span
                  class="nitrozen-option ripple"
                  data-value="false"
                >
                  <div
                    class="nitrozen-option-container"
                  >
                    <span
                      class="nitrozen-option-image"
                    >
                      <!---->
                       Disabled 
                    </span>
                  </div>
                </span>
                <!---->
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div>
    <div
      class="page-error"
    >
      <img
        src="/public/assets/pngs/page-error.png"
      />
       
      <div
        alt="Error"
        class="page-error-text"
      >
        <span
          class="dark-sm cl-DustyGray2"
        >
          Something has gone wrong on!
        </span>
      </div>
       
      <div>
        <button
          class="n-button ripple n-button-secondary n-flat-button n-flat-button-secondary"
          href=""
          type="button"
        >
          <div
            class="n-button-content"
          >
            <!---->
            Try again
    
            <!---->
          </div>
        </button>
      </div>
    </div>
     
    <!---->
  </div>
</div>
`;
